### PR TITLE
UTOPIA-737: Adding Success Notification to Teams Channel

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

UTOPIA-737: Adding Success Notification to Teams Channel

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by akroon3r** [comment]
I've added the TEAMS_SUCCESS_URL environment variable to the global scope of repository secrets. I've also added the TEAMS_FAILURE_URL environment variable for subsequent pull requests to come.

**Review by kushal-arora-fw — APPROVED** [review]
I am assuming all the tokens are residing inside the "TEAMS_SUCCESS_URL"?

File: .github/workflows/build-deploy-api-v2.yaml [hunk 1]
```diff
@@ -161,3 +161,21 @@ jobs:
...
+
+        - name: Message Teams Channel on Success
+          run: |
+            curl --location --request POST ${{ secrets.TEAMS_SUCCESS_URL }} \
+            --header 'Content-Type: text/plain' \
+            --data-raw '{
+              "@context": "https://schema.org/extensions",
+              "@type": "MessageCard",
+              "themeColor": "0072C6",
+              "title": "SUCCESS! Deployment to PIA ${{ secrets.ENVIRONMENT }} Environment has finished.",
```
Comment (antsand) [inline .github/workflows/build-deploy-api-v2.yaml:179]: Does a team message have a title?

File: .github/workflows/build-deploy-api-v2.yaml [hunk 2]
```diff
@@ -161,3 +161,21 @@ jobs:
...
+              "text": "The PIA API has been built and deployed to the ${{ secrets.ENVIRONMENT }} environment. \n\n The logs for the successful deployment can be found (here)[${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}]\n\n The changes for commit ${{ github.sha }} can be found [here](https://dev.pia.gov.bc.ca/api/api-docs).",
```
Comment (antsand) [inline .github/workflows/build-deploy-api-v2.yaml:180]: should this be content? 
https://learn.microsoft.com/en-us/graph/api/channel-post-messages?view=graph-rest-1.0&tabs=http

File: .github/workflows/build-deploy-api-v2.yaml (hunk 1 above)
Comment (akroon3r) [reply .github/workflows/build-deploy-api-v2.yaml:179]: please go check the #deployments channel in the PIA teams channel.

you can see the formatting I've generated and tested already from postman

File: .github/workflows/build-deploy-api-v2.yaml (hunk 2 above)
Comment (akroon3r) [reply .github/workflows/build-deploy-api-v2.yaml:180]: I'm working of the example found here:

https://learn.microsoft.com/en-us/outlook/actionable-messages/send-via-connectors

**Review by binggaofw — APPROVED** [review]

File: .github/workflows/build-deploy-api-v2.yaml (hunk 2 above)
Comment (antsand) [reply .github/workflows/build-deploy-api-v2.yaml:180]: image

I think some of the values are not de-referenced.

**Pull request merged**